lib/puma/server.rb

```diff
@@ -28,6 +28,9 @@ class Server
     include Puma::Const
     extend  Puma::Delegation
 
+    # TCP_TIME_WAIT: 6, TCP_CLOSE: 7, TCP_CLOSE_WAIT: 8, TCP_LAST_ACK: 9
+    IGNORED_CHECK_CLIENT_SOCKET_STATES = (6..9)
```

Comment:
For `TIME_WAIT` to appear it means the application already called `close()` on the socket, which should be impossible here, and I'd also bet it's not legal to call `close()` on such a socket since it implies closing it twice.